src/debug-server.ts

```diff
@@ -411,126 +517,415 @@ export class DebugServer extends EventEmitter implements DebugServerEvents {
         return lines.map((line, i) => `${i + 1}: ${line}`).join('\n');
     }
 
-    private async handleDebug(payload: { steps: DebugStep[] }): Promise<string[]> {
-        const results: string[] = [];
+    private async executeDebugPlan(payload: { steps: DebugStep[] }): Promise<DebugExecutionEnvelope> {
+        const steps = payload.steps || [];
+        const sessionId = crypto.randomUUID();
+        const results: DebugStepResult[] = [];
+        const bufferedEvents: DebugEvent[] = [];
+
+        // Subscribe to debug events
+        const disposables: vscode.Disposable[] = [];
+
+        disposables.push(
+            vscode.debug.onDidReceiveDebugSessionCustomEvent(evt => {
+                bufferedEvents.push({
+                    event: 'debug.custom',
+                    body: evt.body,
+                    sessionId: evt.session.id,
+                    timestamp: nowIso()
+                });
+            }),
+            vscode.debug.onDidTerminateDebugSession(s => {
+                bufferedEvents.push({ 
+                    event: 'debug.terminated', 
+                    sessionId: s.id, 
+                    timestamp: nowIso() 
+                });
+            }),
+            vscode.debug.onDidStartDebugSession(s => {
+                bufferedEvents.push({ 
+                    event: 'debug.started', 
+                    sessionId: s.id, 
+                    timestamp: nowIso() 
+                });
+            })
+        );
```

Comment:
seems like we could get some real value from https://modelcontextprotocol.io/specification/2025-06-18/server/utilities/logging